match-standards-serum-mix01.Rmd

```diff
@@ -0,0 +1,3178 @@
+`r std` from HMDB.
+
+```{r mix01-serum-3-phosphoglyceric-acid-ms2-hmdb-heatmap, fig.cap = "Similarities of MS2 spectra for selected features against reference spectra for 3-Phoshoglyceric Acid from HMDB.", echo = FALSE, fig.width = 5, fig.height = 5}
+hmdb_id <- std_dilution$HMDB[std_dilution$name == std]
+hmdb <- Spectra(cdb, filter = ~ compound_id == hmdb_id)
+
+sim <- compareSpectra(std_ms2, hmdb, ppm = 40)
+
+ann <- data.frame(feature_id = std_ms2$feature_id, rt = rtime(std_ms2))
+rownames(ann) <- rownames(sim)
+
+pheatmap(sim, annotation_row = ann, breaks = seq(0, 1, length.out = 101),
+         color = colorRampPalette((brewer.pal(n = 7, name = "YlOrRd")))(100))
+```
+
+It seems that no extracted MS2 spectra matches well the reference ones from
+3-Phosphoglyceric Acid. The maximum score is around 0.08 and is attained by a
+spectrum from FT02217. The only two matches with score greater than 0.07 are
+shown below.
+
+```{r mix01-serum-3-phosphoglyceric-acid-mirror-hmdb, echo = FALSE, fig.cap = "Mirror plots"}
+std_ms2_sel <- plot_select_ms2(0.07)
+```
+
+Since all the extracted MS2 spectra are associated to ions different from
+[M+H]+ we also perform a comparison between the neutral loss spectra 
+
+```{r mix01-serum-3-phosphoglyceric-acid-nl-ms2-hmdb-heatmap, fig.cap = "Similarities of MS2 neutral loss spectra for selected features against reference neutral loss spectra for 3-Phoshoglyceric Acid from HMDB.", echo = FALSE, fig.width = 5, fig.height = 5}
+nl_hmdb <- all_hmdb_nl_ms2[all_hmdb_nl_ms2$compound_id == hmdb_id]
+nl_std_ms2 <- neutralLoss(std_ms2, PrecursorMzParam())
+sim <- compareSpectra(nl_std_ms2, nl_hmdb, ppm = 40)
+
+ann <- data.frame(feature_id = nl_std_ms2$feature_id, rt = rtime(nl_std_ms2))
+rownames(ann) <- rownames(sim)
+
+pheatmap(sim, annotation_row = ann, breaks = seq(0, 1, length.out = 101),
+         color = colorRampPalette((brewer.pal(n = 7, name = "YlOrRd")))(100))
+```
+
+No match was found. In addition we match (**all**) the MS2 spectra for the
+matched features against all spectra from HMDB identifying reference spectra
+with a similarity larger than 0.7. This is to ensure that no reference spectra
+from any other compound would match the extracted spectra for this feature with
+a high similarity.
+
+```{r}
+hmdb_match <- matchSpectra(
+    std_ms2, all_hmdb_ms2,
+    param = CompareSpectraParam(ppm = 40, requirePrecursor = FALSE))
+hmdb_match
+```
+
+```{r}
+hmdb_match <- matchSpectra(
+    nl_std_ms2, all_hmdb_nl_ms2,
+    param = CompareSpectraParam(ppm = 40, requirePrecursor = FALSE))
+hmdb_match
+```
+
+No spectrum in HMDB matches any of the input spectra.
+We next we compare the experimental MS2 spectra against MassBank.
+
+```{r}
+mbank_match <- matchSpectra(
+    std_ms2, mbank,
+    param = CompareSpectraParam(ppm = 40, requirePrecursor = FALSE))
+mbank_match
+```
+
+```{r}
+# # throws an error. Maybe there's some problem in the packages? 
+# mbank_match <- matchSpectra(
```

Comment:
Here the error message

Reply:
Perfect! Thanks, I'll have a look at it.